### PR TITLE
Fix Amazon Comprehend with SageMaker Pipelines clean up

### PR DESCRIPTION
**Pull request by eugeneteoh**
*Description of changes:*
Fix cleanup scripts for Amazon Comprehend with SageMaker Pipelines example (#2979).
Add clean up scripts for Comprehend endpoint, Lambda function and Lambda IAM role, and added and updated README.

*Testing done:*
SageMaker Studio end to end.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-sagemaker-examples/blob/master/CONTRIBUTING.md) doc and adhered to the example notebook best practices
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-sagemaker-examples/blob/master/README.md)
- [x] I have tested my notebook(s) and ensured it runs end-to-end
- [x] I have linted my notebook(s) and code using `tox -e black-format,black-nb-format`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -190,6 +190,12 @@ These examples show you how to use SageMaker Processing jobs to run data process
 - [Feature transformation with Amazon SageMaker Processing and Dask](sagemaker_processing/feature_transformation_with_sagemaker_processing_dask) shows how to use SageMaker Processing to transform data using Dask distributed clusters
 - [Distributed Data Processing using Apache Spark and SageMaker Processing](sagemaker_processing/spark_distributed_data_processing) shows how to use the built-in Spark container on SageMaker Processing using the SageMaker Python SDK.
 
+### Amazon SageMaker Pipelines
+
+These examples show you how to use [SageMaker Pipelines](https://aws.amazon.com/sagemaker/pipelines) to create, automate and manage end-to-end Machine Learning workflows.
+
+- [Amazon Comprehend with SageMaker Pipelines](sagemaker-pipelines/nlp/amazon_comprehend_sagemaker_pipeline) shows how to deploy a custom text classification using Amazon Comprehend and SageMaker Pipelines.
```
Comment (jkroll-aws) [inline README.md:197]: "a custom text classification" sounds incomplete to me - how about "a custom text classification model"? Occurs in both READMEs.

File: sagemaker-pipelines/nlp/amazon_comprehend_sagemaker_pipeline/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+[sm_pipeline_with_comprehend.ipynb](sm_pipeline_with_comprehend.ipynb): Notebook explaining the pipeline step-by-step.
+
+[prepare_data.py](prepare_data.py): Script used in ComprehendProcess step in pipeline for data preparation used for training and testing.
+
+[train_eval_comprehend.py](train_eval_comprehend.py): Script used in ComprehendTrainAndEval step in pipeline to train and evaluate the Amazon Comprehend model.
+
+[deploy_comprehend.py](deploy_comprehend.py): Script used in ComprehendDeploy step in pipeline to deploy an Amazon Comprehend model endpoint.
+
+[iam_helper.py](iam_helper.py): Helper function to create IAM for lambda function.
```
Comment (jkroll-aws) [inline sagemaker-pipelines/nlp/amazon_comprehend_sagemaker_pipeline/README.md:15]: Nits: Be more specific than "create IAM" - accurate to say "create an IAM role"? Capitalize `Lambda`.

Comment (eugeneteoh) [reply sagemaker-pipelines/nlp/amazon_comprehend_sagemaker_pipeline/README.md]: Fixed

File: README.md (hunk 1 above)
Comment (eugeneteoh) [reply README.md:197]: Fixed

**Review by jkroll-aws — APPROVED** [review]

**Pull request merged**